### DISCUSSION TITLE
Can recharts be used with Angular?

### DISCUSSION THREAD

**Discussion by vishaalsharmaa** [recharts/recharts#2767] [Q&A]
Can I use recharts with Angular?

**Comment by harshagarwalnyu** [comment]
No - Recharts is React-specific. It renders via React components and depends on React's virtual DOM and lifecycle hooks, so it cannot be used in Angular projects.

**Angular-native alternatives:**

| Library | Notes |
|---------|-------|
| [ngx-charts](https://github.com/swimlane/ngx-charts) | Closest API feel to Recharts; uses D3 internally, fully Angular |
| [ng2-charts](https://github.com/valor-software/ng2-charts) | Angular wrapper around Chart.js; large ecosystem |
| [ngx-echarts](https://github.com/xieziyu/ngx-echarts) | Angular wrapper for Apache ECharts; best for complex/interactive charts |
| [Highcharts Angular](https://github.com/highcharts/highcharts-angular) | Official Highcharts wrapper; commercial license for non-OSS |
| [D3.js](https://d3js.org/) | Framework-agnostic; highest flexibility, steeper learning curve |

**ngx-charts** is the closest in spirit to Recharts - declarative, composable chart components with built-in animations. If you're migrating from Recharts, that's the smoothest path.
